File: 프로그래머스/unrated/181911. 부분 문자열 이어 붙여 문자열 만들기/부분 문자열 이어 붙여 문자열 만들기.cs
using System;
using System.Text;

public class Solution {
    public string solution(string[] my_strings, int[,] parts) 
    {
        var sb = new StringBuilder();
        for(int i = 0; i < parts.GetLength(0); ++i)
        {
            int start = parts[i, 0];
            int end = parts[i, 1];
            
            string subString = my_strings[i].Substring(start, end - start + 1);
            sb.Append(subString);
        }
        return sb.ToString();
    }
}

File: 프로그래머스/0/181911. 부분 문자열 이어 붙여 문자열 만들기/부분 문자열 이어 붙여 문자열 만들기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(vector<string> my_strings, vector<vector<int>> parts) {
    string answer = "";
    
    for (int i = 0; i < my_strings.size(); ++i)
    {
        string str = my_strings[i];
        int s = parts[i][0];
        int e = parts[i][1];
        
        answer += str.substr(s, e - s + 1);
    }
    
    return answer;
}